precompiles/parachain-staking/StakingInterface.sol

```diff
@@ -1,71 +1,113 @@
+/// @author The Moonbeam Team
+/// @title The interface through which solidity contracts will interact with Parachain Staking
 /// We follow this same interface including four-byte function selectors, in the precompile that
 /// wraps the pallet
 interface ParachainStaking {
     // First some simple accessors
 
-    /// Check whether the specified address is currently a staking nominator
+    /// @dev Check whether the specified address is currently a staking nominator
+    /// @param nominator the address that we want to confirm is a nominator
+    /// @return A boolean confirming whether the address is a nominator
     function is_nominator(address nominator) external view returns (bool);
 
-    /// Check whether the specified addess is currently a collator candidate
+    /// @dev Check whether the specified address is currently a collator candidate
+    /// @param collator the address that we want to confirm is a collator andidate
+    /// @return A boolean confirming whether the address is a collator candidate
     function is_candidate(address collator) external view returns (bool);
 
-    /// Check whether the specifies address is currently a part of the active set
+    /// @dev Check whether the specifies address is currently a part of the active set
+    /// @param collator the address that we want to confirm is a part of the active set
+    /// @return A boolean confirming whether the address is a part of the active set
     function is_selected_candidate(address collator)
         external
         view
         returns (bool);
 
-    /// Total points awarded to all collators in a particular round.
+    /// @dev Total points awarded to all collators in a particular round
+    /// @param round the round for which we are querying the points total
+    /// @return The total points awarded to all collators in the round
     function points(uint256 round) external view returns (uint256);
 
-    /// Get the minimum nomination amount
+    /// @dev Get the minimum nomination amount
+    /// @return The minimum nomination amount
     function min_nomination() external view returns (uint256);
 
+    /// @dev Get the CandidateCount weight hint
+    /// @return The CandidateCount weight hint
+    function candidate_count() external view returns (uint256);
+
+    /// @dev Get the CollatorNominationCount weight hint
+    /// @param collator The address for which we are querying the nomination count
+    /// @return The number of nominations backing the collator
+    function collator_nomination_count(address collator)
+        external
+        view
+        returns (uint256);
+
+    /// @dev Get the NominatorNominationCount weight hint
+    /// @param nominator The address for which we are querying the nomination count
+    /// @return The number of nominations made by the nominator
+    function nominator_nomination_count(address nominator)
+        external
+        view
+        returns (uint256);
+
```

Comment:
Would these be possible to grab by querying the chain state via the polkadot js api, say from the `candidatePool` function? Even if so, seems like a roundabout way to do it. It makes sense to me that this should be part of the precompile.

Reply:
It is possible, but, as you said, this requires using polkadot-js API. The point of putting it in the precompile interface is to enable metamask to access this information (which is how the Dapp will work).